src/main/kotlin/com/amazon/opendistroforelasticsearch/indexmanagement/indexstatemanagement/elasticapi/ElasticExtensions.kt

```diff
@@ -188,18 +188,17 @@ fun IndexMetadata.shouldDeleteManagedIndexMetaData(): Boolean =
  * Returns the current policy_id if it exists and is valid otherwise returns null.
  * */
 fun IndexMetadata.getPolicyID(): String? {
-    if (this.settings.get(ManagedIndexSettings.POLICY_ID.key).isNullOrBlank()) return null
-
-    return this.settings.get(ManagedIndexSettings.POLICY_ID.key)
+    if (ManagedIndexSettings.POLICY_ID.get(this.settings).isNullOrBlank()) return null
+    return ManagedIndexSettings.POLICY_ID.get(this.settings)
 }
 
 /**
  * Returns the current rollover_alias if it exists otherwise returns null.
  * */
 fun IndexMetadata.getRolloverAlias(): String? {
-    if (this.settings.get(ManagedIndexSettings.ROLLOVER_ALIAS.key).isNullOrBlank()) return null
```

Comment:
I feel this way is more straightforward

Reply:
Can change it back, was originally changed because I had this setting deprecated with a new setting name and this old way would not be able to default back to deprecated value.